### PR TITLE
Update models.cljs

### PR DESCRIPTION
**Pull request by krisc**
No longer redirects to home tab when qr-code error message is dismissed.

Fixes issue https://github.com/status-im/status-react/issues/5287

### CODE REVIEW

File: src/status_im/ui/screens/add_new/models.cljs [hunk 1]
```diff
@@ -12,11 +12,8 @@
     (universal-links/handle-view-profile cofx data)
     (or (universal-links/handle-url cofx data)
         {:utils/show-popup {:title      (i18n/label :t/unable-to-read-this-code)
-                            :content    (i18n/label :t/use-valid-qr-code {:data data})
-                            :on-dismiss #(re-frame/dispatch [:navigate-to-clean :home])}})))
+                            :content    (i18n/label :t/use-valid-qr-code {:data data})}})))
 
 (fx/defn handle-qr-code
```
Comment (flexsurfer) [inline src/status_im/ui/screens/add_new/models.cljs]: so looks like there is no need to have `handle-qr-code` at all, we can use  `process-qr-code` directly

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/ui/screens/add_new/models.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/ui/screens/add_new/models.cljs]: @krisc Can you comment on that?

Comment (rasom) [reply src/status_im/ui/screens/add_new/models.cljs]: fixed

**Review by rasom — APPROVED** [review]

**Comment by asemiankevich** [comment]
@krisc when i dismiss the error message - camera is closed and i am on the home tab. Could you point what exactly was changes and how it reflects in your PR?

Here is the expected output:

#### Expected behavior
* [ ]  do not redirect a user to Home tab after the error dialog for scanning invalid contact code is closed
* [ ]  have the camera opened to the user can scan any other code when the error dialog is closed

File: src/status_im/ui/screens/add_new/models.cljs (hunk 1 above)
Comment (krisc) [reply src/status_im/ui/screens/add_new/models.cljs]: Great. Makes sense just to make it just one function if one pretty much only calls the other.

**Comment by krisc** [comment]
@asemiankevich hmm that's strange. It works as expected for me. I removed the lines of code which returns the user to home screen

**Comment by asemiankevich** [comment]
> @asemiankevich hmm that's strange. It works as expected for me. I removed the lines of code which returns the user to home screen

@krisc i took last ios build https://i.diawi.com/zeykHc from this PR, i tapped + then strat new chat-> and scanned invalid code. I received an error. I closed an error. Result: i am on home tab , camera is off.

<blockquote><div><strong><a href="https://i.diawi.com/zeykHc">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by krisc** [comment]
@asemiankevich I wonder if it has to do with the iOS build. I only tested on my Android device. Has anyone else tried running this build?

**Comment by asemiankevich** [comment]
> @asemiankevich I wonder if it has to do with the iOS build. I only tested on my Android device. Has anyone else tried running this build?

android behaves the same as iOS. After i tap OK at the error dialog - i see Home tab.

**Comment by krisc** [comment]
@asemiankevich can you give me a link to that android build? I am actually not familiar with the Diawi builds. Are those daily builds? I'm just building from source.

**Comment by asemiankevich** [comment]
> @asemiankevich can you give me a link to that android build? I am actually not familiar with the Diawi builds. Are those daily builds? I'm just building from source.

here is last android https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190211-080656-042da0-pr.apk

<blockquote></blockquote>

**Comment by krisc** [comment]
@asemiankevich okay, you're right. this build doesn't work as expected. In fact, even valid QR codes aren't working as expected for me.

Can someone fill me in on what these builds are? Are they daily builds or what? I'm convinced that my commits aren't even part of this build.

**Comment by asemiankevich** [comment]
> @asemiankevich okay, you're right. this build doesn't work as expected. In fact, even valid QR codes aren't working as expected for me.
> 
> Can someone fill me in on what these builds are? Are they daily builds or what? I'm convinced that my commits aren't even part of this build.

the builds are created by jenkins right from your branch. The table with the builds present in this pull request item.

https://github.com/status-im/status-react/pull/7449#issuecomment-461574791

**Comment by krisc** [comment]
Hmm okay. Then I wonder why it behaves as expected when I build from source

**Comment by krisc** [comment]
Oh, scratch that! The build you linked to me does in fact behave as expected. I was testing the wrong thing. How I'm testing is:

1. Tap '+' on home screen.
2. Select 'Scan QR code'
3. Scan an invalid QR code.
4. Then an error pops up "Unable to read this code ... " And when I hit 'OK' it remains in the QR scanner.

@asemiankevich not sure why it is behaving differently for you.

**Comment by krisc** [comment]
Maybe someone else can try running this build and see what's going on here? @rasom @flexsurfer @jeluard

**Comment by asemiankevich** [comment]
@Serhy @churik  please assist

**Comment by churik** [comment]
@asemiankevich @krisc 
Tested on [last IOS](https://i.diawi.com/zeykHc) and [last android](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190211-080656-042da0-pr.apk) builds.
For me when I dismiss the error message - camera is closed and i am on the home tab, so issue is still there.

Devices: IOS 11.4, IPhone 7; Android 7.0, LG V20.

**Comment by krisc** [comment]
@asemiankevich @churik That's really weird. I tested it again from that build you linked to me, and it works as expected. I am using Android 7.0 too.

**Comment by krisc** [comment]
@asemiankevich @churik  here's a video https://d.tube/v/kuyakris/lej2o2tl

<blockquote><div> DTube</div><div><strong><a href="https://d.tube/#!/v/kuyakris/lej2o2tl">20190214_081624</a></strong></div><div>for status team</div></blockquote>

**Comment by Serhy** [comment]
@krisc I can confirm too that issue is present for me as well as for @asemiankevich and @churik . And at the same time, I can confirm it's OK on my side your scenario too :) (almost OK, please see below)

Following description from original issue (#5287):
* Open Status
* Tap plus button -> start new chat _(please note it's very first item in the list)_
* Tap scan button _(it's the QR code button on the right side from 'Enter contact code or username' field)_
* Scan invalid QR (that does not contain valid contact code)
* Pay attention `Unable to read this code` dialog appears
* Tap ok to close the dialog
**Actual resut:** User is redirected to Home screen
**Expected result:**  User have the camera opened
* Now, having the camera opened scan valid QR code
**Expected result:** QR code scanned and user wind up on the contact profile screen 

But indeed, as you described above https://github.com/status-im/status-react/pull/7449#issuecomment-463360985 - OK on my side, yes.

With the `Scan QR code` you've made a change to, however, the only thing is that after incorrect contact code was scanned followed by the `Unable to read this code` pop-up appeared, - nothing happens when I'm trying to scan 'valid' contact code (That is what second item from description is about: i.e. 'Have the camera opened to the user can scan any other code when the error dialog is closed' is not happening for me.

**Comment by krisc** [comment]
@Serhy thank you for taking a look. That's an obvious part of the expected behavior that I suppose I overlooked. Consider this a WIP then. I'll look into this.

I still don't understand how @asemiankevich and @churik are having their issues though.

**Comment by Serhy** [comment]
@krisc well, @asemiankevich and @churik can reproduce the issue as well as I can reproduce it too when following scenario from the description #5287 :

* Open Status
* Tap plus button -> **Start new chat** _(please note it's very first item in the list)_
* Tap scan button _(it's the QR code button on the right side from 'Enter contact code or username' field)_
* Scan invalid QR (that does not contain valid contact code)
* Pay attention `Unable to read this code` dialog appears
* Tap ok to close the dialog
  **Actual resut:** User is redirected to Home screen
  **Expected result:**  User have the camera opened
* Now, having the camera opened scan valid QR code
  **Expected result:** QR code scanned and user wind up on the contact profile screen

There are two ways at this moment to add new contact via QR code scan withing Status app: 
- (1) Being on Home screen tap plus button -> tap **Start new chat** -> tap square button next to contact code editbox 
- (2) Being on Home screen tap plus button -> tap **Scan QR code**

With (1) we can reproduce issue.
With (2) it looks OK (except the `Have the camera opened to the user can scan any other code when the error dialog is closed' is not happening for me` part)

**Comment by krisc** [comment]
@Serhy oh I see. I only considered case (2). Thanks!

**Comment by krisc** [comment]
@Serhy  @asemiankevich @churik 
 Okay that latest commit should fix those issues in both cases. Let me know if I overlooked something.

**Comment by churik** [comment]
@krisc 
issue mentioned is fixed, thanks!
One more regression issue is found (reproducible on IOS): back button on "Scan QR" code screen can lead to white screen and generally it doesn't redirect back to "New Contact" screen

**Scenario 1 (steps):**
* Open Status
* Tap plus button -> **Start new chat** _(please note it's very first item in the list)_
* Tap scan button _(it's the QR code button on the right side from 'Enter contact code or username' field)_
* Try to tap Back _(it's not working from the first time in your PR, so try tap several times in a row)_

**Scenario 2(steps):**
* Open Status
* Tap plus button -> **Scan QR code**
* Try to tap Back _(it's not working from the first time in your PR, so try tap several times in a row)_

**Expected result:**  can go back easily  ([behavior in nightly 16/02/19](https://monosnap.com/file/YOWVLJrPdrVaxIyFvtO23jh7ZSgosr))

**Actual result:**  back button doesn work and if you trying to tap several times in a row, it leads to white screen ([video from PR build](https://take.ms/apDLI) )

Link to build: https://i.diawi.com/y2PdkL

<blockquote><div><strong><a href="https://i.diawi.com/y2PdkL">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by krisc** [comment]
@churik is that due to my change, or is this a separate issue?

**Comment by churik** [comment]
@krisc it is due to your changes because it is working in current nightly

**Comment by krisc** [comment]
> @krisc it is due to your changes because it is working in current nightly

I think I know why it might be doing that. I'll look into this today.

**Comment by krisc** [comment]
@churik Okay, I think what's going on is on dismissal of the error message, I call to `:navigate-to-clean` to go back to the QR scanner and allow the user to scan another code. But at the same time, `:navigate-to-clean` QR scanner doesn't allow back navigation. Not sure what to do about this yet so I'm open to suggestions, but I will get back to this later.

**Comment by churik** [comment]
@krisc in any case back navigation should be presented on this screen IMO. So looking forward for fix, I can't say that it is possible to choose between https://github.com/status-im/status-react/issues/5287 and back navigation.
Thank you for investigation!

**Comment by krisc** [comment]
@churik so can we have both? I can continue with this if someone can help me with that. Also okay with handing this bounty off to someone else.

**Comment by churik** [comment]
@krisc I don't think that we have any limitation for the number of contributors per issue.

**Comment by pablanopete** [comment]
@yenda @rasom or @flexsurfer could you help with @krisc on this issue? He's needing a little direction on why his fix is causing this regression as above. "One more regression issue is found (reproducible on IOS): back button on "Scan QR" code screen can lead to white screen and generally it doesn't redirect back to "New Contact" screen". 

A newcomer to our codebase who is trying his hardest to learn the ropes, so a little guidance would go a long way! Thanks.

**Comment by yenda** [comment]
@krisc if you want we can have a call and I can give you a walkthrough of the codebase. You can also add me on status. Just reading the thread I'd say you might want to use navigate-to-cofx instead of clean, I'll have a deeper look tomorrow

**Comment by krisc** [comment]
@yenda I'd be up for a call if our hours allow for it! I'm in PST timezone and usually free from 8am to 3pm PST. In any case, please send me a message on status 
https://get.status.im/user/0x04825abbccf20caa211e1564dc566222278f3f08a5d0013f9277d643a8e10e52e48486248d432a31ce3832ce7eb51d44a62503e54f8088595310b28c0a2a3016c3

<blockquote><div><strong><a href="https://get.status.im/user/0x04825abbccf20caa211e1564dc566222278f3f08a5d0013f9277d643a8e10e52e48486248d432a31ce3832ce7eb51d44a62503e54f8088595310b28c0a2a3016c3">View user 0x04825abbccf20caa211e1564dc566222278f3f08a5d0013f9277d643a8e10e52e48486248d432a31ce3832ce7eb51d44a62503e54f8088595310b28c0a2a3016c3 profile in Status</a></strong></div></blockquote>

**Comment by krisc** [comment]
@yenda did you have a chance to look at this yet?

I looked into `navigate-to-cofx` and noticed there isn't an event handler for it in `src/status-im/ui/screens/navigation.cljs`. Would a new event handler have to be made so it can be used like this?

```
:on-dismiss #(re-frame/dispatch [:navigate-to-cofx :qr-scanner cofx])}})))
```

**Comment by yenda** [comment]
@krisc we can discuss that during the call, but even if it is not respected everywhere in the code, the convention is to name events after what happened, and to compose fx functions together. The fx functions are the functions defined with fx/def and the composition is done with fx/merge. So in your case you can create an event in events.cljs and use navigate-to-cofx there. We can discuss this things on the call if you want, when are you available this week? I'm CEST timezone so I guess the best time is 8am your time

**Comment by krisc** [comment]
@yenda this week Tuesday and Thursday morning would work best for me.

**Comment by yenda** [comment]
@krisc I added you on status we can chat there

**Comment by jeluard** [comment]
@krisc Could you make progress and discuss with @yenda ?

**Comment by krisc** [comment]
@jeluard oh, yes we are in touch and I am making progress

**Comment by krisc** [comment]
please see this PR https://github.com/status-im/status-react/pull/7745

**Comment by churik** [comment]
https://github.com/status-im/status-react/pull/7745 is merged, so this one can be closed afaik.

**Pull request closed without merging**